code/game/atoms_movable.dm

```diff
@@ -69,6 +73,11 @@
 	. = ..()
 	if(loc)
 		loc.handle_atom_del(src)
+	if(length(del_subscribers))
+		for(var/atom/sub in del_subscribers)
+			if(length(del_subscribers[sub])) // a check that the subscriber is still actually subscribed.
+				sub.handle_atom_del(src)
```

Comment:
So you plan to handle GC issues... with more hard refs?

Reply:
Why do you think i put the PR up as draft, said it's not ready for review, and posted the GC log demonstrating exactly that?

Fixed, they are no longer hard refs.